connectd/multiplex.c

```diff
@@ -1111,6 +1112,18 @@ static struct io_plan *read_body_from_peer_done(struct io_conn *peer_conn,
        if (!peer->dev_read_enabled)
 	       return read_hdr_from_peer(peer_conn, peer);
 
+       switch (dev_inbound(&peer->id, type)) {
+       case DEV_DISCONNECT_DROP:
+	       /* try again? */
+	       return read_hdr_from_peer(peer_conn, peer);
```

Comment:
This is strictly impossible, since our connections always send in order.  So now we ignore a msg, that contract is broken.  You're using it in a very specific way, so that's OK, but "don't use this" is probably needed.

Can you do it on the send side (i.e. don't send this msg)?  Would be significant code reduction...